aten/src/ATen/native/ComparisonUtils.cpp

```diff
@@ -0,0 +1,36 @@
+#include <ATen/core/TensorBase.h>
+#include <algorithm>
+#include <vector>
+#include <ATen/core/TensorBody.h>
+
+namespace at {
+
+class Tensor;
+
+namespace native {
+
+void _assert_true(bool value, c10::basic_string_view<char> message) {
+  AT_ASSERT(value, message);
+}
+
+bool _tensor_equal(at::Tensor const& tensor, c10::OptionalArrayRef<long> sizes, c10::OptionalArrayRef<long> strides, c10::optional<c10::ScalarType> dtype) {
```

Comment:
This name might be confusing with `__eq__`, perhaps `metadata_equal`

Reply:
Sure